nighthawk/include/nighthawk/client/worker.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <memory>
+
+namespace Nighthawk {
+namespace Client {
+
+/**
+ * Interface for a threaded benchmark client worker. All routines are thread safe.
+ */
+class Worker {
+public:
+  virtual ~Worker() {}
+
+  /**
+   * Start the worker thread.
+   */
+  // TODO(oschaaf): Guarddog?
```

Comment:
I don't think we need this for a load generator. We can probably rely on surrounding infrastructure to kill rogue instances.